dialer.go

```diff
@@ -288,9 +315,14 @@ func (d *Dialer) Dial(ctx context.Context, icn string, opts ...DialOption) (conn
 		go trace.RecordDialError(context.Background(), icn, d.dialerID, err)
 		endDial(err)
 	}()
-	cn, err := instance.ParseConnName(icn)
+	cn, err := d.resolveInstanceName(ctx, icn)
 	if err != nil {
-		return nil, err
+		// The connection name was not project:region:instance
```

Comment:
I think we don't need this block of code since it's part of `resolveInstanceName`?